### PR TITLE
Creature: Add SetMulticlassLimit

### PR DESCRIPTION
**Pull request by hendrikgit**
Adds `NWNX_Creature_SetMulticlassLimit(object oCreature, int nLimit)` to limit the number of classes a PC can take dynamically.

### CODE REVIEW

File: Plugins/Creature/Creature.cpp [hunk 1]
```diff
@@ -3584,3 +3586,56 @@ NWNX_EXPORT ArgumentStack GetMaxAttackRange(ArgumentStack&& args)
 
     return 0.0f;
 }
+
+NWNX_EXPORT ArgumentStack SetMulticlassLimit(ArgumentStack&& args)
+{
+    if (!s_bGetIsClassAvailableHookInitialized)
```
Comment (Daztek) [inline Plugins/Creature/Creature.cpp:3592]: No need to check/set a static bool, the static hook below will only be initialized once

File: Plugins/Creature/Creature.cpp [hunk 2]
```diff
@@ -51,10 +51,12 @@ static bool s_bCriticalMultiplierHooksInitialized = false;
 static bool s_bCriticalRangeHooksInitialized = false;
 static bool s_bResolveAttackRollHookInitialized = false;
 static bool s_bResolveInitiativeRollHookInitialized = false;
+static bool s_bGetIsClassAvailableHookInitialized = false;
 static Hooks::Hook s_GetClassLevelHook = nullptr;
 
 static std::unordered_map<uint8_t, std::unordered_map<ObjectID, int16_t>> s_RollModifier;
 static std::unordered_map<ObjectID, bool> s_ParryAllAttacks;
+static std::unordered_map<OBJECT_ID, uint> s_MulticlassLimits;
```
Comment (Daztek) [inline Plugins/Creature/Creature.cpp:59]: Why an unordered_map instead of using NWNX's POS? With POS you can have it persist to the bic too

**Comment by hendrikgit** [comment]
Thanks Daz. I removed the static bool and switched to using POS.

**Comment by Daztek** [comment]
Maybe add a `GetMulticlassLimit` too?

**Comment by hendrikgit** [comment]
Added a get function.

**Pull request merged**